api/handlers/shared/status.go

```diff
@@ -43,45 +42,37 @@ func Status(c *websocket.Conn) {
 		}
 	}
 
-	sts := &appsv1.StatefulSet{}
-	stsKey := types.NamespacedName{
-		Namespace: c.Locals("namespace").(string),
-		Name:      c.Params("name"),
-	}
+	ns := c.Locals("namespace").(string)
+	name := c.Params("name")
+	selector := fmt.Sprintf("app.kubernetes.io/managed-by=kotal-operator,app.kubernetes.io/instance=%s", name)
 
-	pod := &corev1.Pod{}
-	key := types.NamespacedName{
-		Namespace: c.Locals("namespace").(string),
-		Name:      fmt.Sprintf("%s-0", c.Params("name")),
+	watch, err := k8sClientset.CoreV1().Pods(ns).Watch(context.Background(), metav1.ListOptions{
+		LabelSelector: selector,
+	})
+	if err != nil {
+		return
 	}
 
-	for {
-
-		err := k8sClient.Get(context.Background(), key, pod)
-		if err != nil {
-
-			// is the pod error due to sts has been deleted ?
-			stsErr := k8sClient.Get(context.Background(), stsKey, sts)
-			if apierrors.IsNotFound(stsErr) {
-				return
-			}
-
-			if apierrors.IsNotFound(err) {
-				err = errors.New("NotFound")
-			}
+	for event := range watch.ResultChan() {
 
-			if err := c.WriteMessage(websocket.TextMessage, []byte(err.Error())); err != nil {
-				return
-			}
-
-			time.Sleep(3 * time.Second)
-			continue
+		pod, ok := event.Object.(*corev1.Pod)
+		if !ok {
+			return
 		}
 
 		phase := string(pod.Status.Phase)
 
 		if pod.DeletionTimestamp != nil {
 			phase = "Terminating"
+
+			// if pod is being terminated, check owner sts is found or not
+			go func() {
+				time.Sleep(3 * time.Second)
+				_, err := k8sClientset.AppsV1().StatefulSets(ns).Get(context.Background(), "my-dot-node", metav1.GetOptions{})
```

Comment:
remove static name my-dot-node and replace it with the variable name

Reply:
Oops thanks for the catch